core/src/main/java/jenkins/cli/StopBuildsCommand.java

```diff
@@ -0,0 +1,135 @@
+            } else if (item != null) {
+                resultBuilder.append(String.format("Cannot abort runs for %s. Unsupported job type: %s\n", item, item.getClass()));
+            } else {
+                resultBuilder.append(String.format("Job with name %s not found.\n", jobName));
+            }
+        }
+
+        stdout.print(resultBuilder.toString());
+        return 0;
+    }
+
+    private void stopJobBuilds(final Job job,
+                               final String jobName,
+                               final StringBuilder resultBuilder) throws IOException, ServletException {
+        final Run lastBuild = job.getLastBuild();
+        final List<String> stoppedBuildsNames = new ArrayList<>();
+        if (lastBuild != null) {
+            stopBuild(lastBuild, jobName, stoppedBuildsNames);
+            checkAndStopPreviousBuilds(lastBuild, jobName, stoppedBuildsNames);
+        }
+        updateResultOutput(resultBuilder, jobName, stoppedBuildsNames);
+    }
+
+    private void stopBuild(final Run build,
+                           final String jobName,
+                           final List<String> stoppedBuildNames) throws IOException, ServletException {
+        if (build.isBuilding()) {
+            final String buildName = build.getDisplayName();
+            stoppedBuildNames.add(buildName);
+            Executor executor = build.getExecutor();
+            if (executor != null) {
+                executor.doStop();
+                logBuildStopped(jobName, buildName);
+            } else {
+                LOGGER.log(Level.INFO, String.format("Build %s in job %s not stopped", buildName, jobName));
+            }
+        }
+    }
+
+    private void checkAndStopPreviousBuilds(final Run lastBuild,
+                                            final String jobName,
+                                            final List<String> stoppedBuildsNames) throws IOException, ServletException {
+        Run build = lastBuild.getPreviousBuildInProgress();
+        while (build != null) {
+            stopBuild(build, jobName, stoppedBuildsNames);
+            build = build.getPreviousBuildInProgress();
+        }
+    }
+
+    private void updateResultOutput(final StringBuilder result,
+                                    final String jobName,
+                                    final List<String> stoppedBuildNames) {
+        if (stoppedBuildNames.isEmpty()) {
+            result.append(String.format("No builds stopped for job '%s'", jobName));
+        } else {
+            result.append(String.format("Builds stopped for job '%s': ", jobName));
+            for (String buildName : stoppedBuildNames) {
+                result.append(buildName).append("; ");
+            }
+        }
+        result.append("\n");
+    }
+
+    private void logBuildStopped(final String jobName, final String buildName) {
+        LOGGER.log(Level.INFO, String.format("Build %s in job %s aborted", buildName, jobName));
```

Comment:
Probably unnecessary. Jenkins already logs a message when a build terminates for any reason.